synapseclient/core/utils.py

```diff
@@ -1214,3 +1217,49 @@ def print_tick(self):
             sys.stdout.write(f"\r {spinner} {self.msg}")
             sys.stdout.flush()
         self._tick += 1
+
+
+def otel_trace_method(
+    method_to_trace_name: typing.Optional[typing.Callable[..., str]] = None
+):
+    """
+    Decorator to trace a method with OpenTelemetry in a sync environment. This function
+    is specifically written to be used on a method within a class.
+
+    This will pass the class instance as the first argument to the method. This allows
+    you to modify the name of the trace to include information about the class instance.
+
+    Example: Decorating a method within a class that will be traced with OpenTelemetry.
+        Setting the trace name with a lambda function:
+
+            @otel_trace_method(method_to_trace_name=lambda self, **kwargs: f"Store: {self.name}")
+            def store(self):
```

Comment:
Nice, do we want to include the `opentelemetry_context` parameter here to let people know about concurrent operations?

Reply:
Added more context in https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1055/commits/a38f2c2bfa771afdd0dad03a9382bce311b98ce4